pkg/util/tracing/service/service.go

```diff
@@ -49,8 +49,13 @@ func (s *Service) GetSpanRecordings(
 		if span.TraceID() != request.TraceID {
 			return nil
 		}
+		var recording tracing.Recording
 		for _, rec := range span.GetRecording() {
-			resp.SpanRecordings = append(resp.SpanRecordings, rec)
+			recording = append(recording, rec)
```

Comment:
span.GetRecording() already returns a tracing.Recording, shouldn't we be able to simply append that to resp.Recordings instead of iterating over every RecordedSpan?

Reply:
doh nice catch.